### ISSUE TITLE
vivaldi windows don't focus properly

### ISSUE THREAD

**Issue by classabbyamp** [baskerville/bspwm#1105]
I have 2 monitors, with several desktops on each. Let's say 1, 2, 3 are on monitor 1 and 4, 5, 6 are on monitor 2. Monitor 2 is my primary monitor.

The first-to-open vivaldi window is on desktop 4. I have a couple other vivaldi windows on desktop 1.

When I switch to desktop 5 from 4, where I am working on something else, then try to mouse over to my vivaldi windows on desktop 1, all keyboard shortcuts and input act on the window on desktop 4.

This doesn't happen when I move my mouse from desktop 4 to 1.

my `.bspwmrc`:
    
    $HOME/.config/polybar/launch.sh &
    
    bspc monitor DP-0 -d 1 2 3
    bspc monitor DP-2 -d 4 5 6
    
    bspc config focus_follows_pointer true
    
    bspc config border_width         2
    bspc config window_gap           10
    
    bspc config split_ratio          0.5
    bspc config borderless_monocle   true
    bspc config gapless_monocle      true

**Comment by plitter** [comment]
Does it work on other windows or exclusively to vivaldi? Maybe `bspc config ignore_ewmh_focus on` helps?

**Comment by classabbyamp** [comment]
I've only had this problem with vivaldi. That config option doesn't do anything.

**Comment by SeerLite** [comment]
I have the same problem, but I'm pretty sure it's Vivaldi's fault and not bspwm's. Have you tested it with other window managers?

**Comment by classabbyamp** [comment]
I've not tried it with other WMs

**Comment by SeerLite** [comment]
[Here's a relevant VIvaldi Forums post](https://forum.vivaldi.net/topic/43425/keyboard-shortcuts-in-wrong-window/2). The OP did not state their OS there. Maybe it even happens on Windows too?

I have also been having this problem for quite a while. Never bothered to fix it though

**Comment by classabbyamp** [comment]
> Current version: www-client/vivaldi-2.10.1745.23_p1 (**gentoo**)

I have vivaldi on a windows computer too, but I didn't notice anything there I think

**Comment by SeerLite** [comment]
I just tested this out. Turns out I don't have this problem anymore. What version of Vivaldi are you running? Maybe it was fixed recently.
```
$ vivaldi-stable --version
Vivaldi 3.0.1874.33
```

**Comment by classabbyamp** [comment]
I'm on `Vivaldi 2.11.1811.49` now, I should probably update, but I was about to switch some stuff up on this machine so I haven't bothered.

**Comment by SeerLite** [comment]
Yeah, I think updating might be the solution.
It's still strange though. There's a person running Arch + bspwm (same setup as me) on that forum post who had this problem yesterday.

**Comment by SeerLite** [comment]
Nevermind, I'm still having this issue. It's pretty random, it doesn't happen all the time.

**Comment by classabbyamp** [comment]
Yep, just checked on Arch with `Vivaldi 3.0.1874.38`, is still there

**Comment by SeerLite** [comment]
Ok so for the record, this can be reproduced without an external monitor:

1. Open Vivaldi on a desktop (let's say desktop `1`)
2. Open a new Vivaldi window ("Ctrl + N" assuming default keybinds).
3. Move the new window to a different desktop (desktop `2` for example).
4. Execute a keybind (like "Ctrl + T").
5. If the keybind works fine, switch to the first desktop (`1`), focusing the original Vivaldi window, and then back to the second (`2`). The keybind should now react on the first window (on `1`) instead of the focused one (on `2`).

I could not reproduce this in i3.

**Comment by SeerLite** [comment]
Note that I have `ignore_ewmh_focus` set to `true`. If I disable it, the original Vivaldi window also steals focus when reacting to the keybinds.

**Comment by plitter** [comment]
I tested and I also have the same issue. But it is interesting for me since when the ctrl-T works on the other vivaldi, if I start my window chooser it starts on the desktop with the vivaldi that is not currently visible! This isn't the first time it has happened either, suddenly some of my window chooser terminal windows don't show up on the screen I'm on, but they'll be responding to choice that I'll make. So I think that in the background vivaldi has changed focus to the other window, and when I open a program, it opens on the desktop that has "focus", not the desktop that is visible, but on the desktop has the currently focused window. I'm betting vivaldi devs could shed some light on how this works.

**Comment by deadlysyn** [comment]
+1 same issue, one monitor, Arch, vivaldi 3.4.2066-94-4 from AUR.

Doesn't matter which desktop I start on, if i manually focus vivaldi on current desktop, ignore ewmh, etc. commands (CTRL-F, CTRL-T, CTRL-W, etc) always target the first window opened.

In fact, to start I found I didn't even get window border...but changing browser option to use "native windows" fixed that. Never had similar issues in firefox, ungoogled-chromium or brave. I've considered just changing back to one of those and avoiding this problem, but in basemark and webxprt vivaldi blew the others away so I'm suffering for premature optimization. :-P
